sympy/tensor/array/ndim_array.py

```diff
@@ -320,25 +319,6 @@ def f(sh, shape_left, i, j):
     def __repr__(self):
         return self.__str__()
```

Comment:
Same here and with the other non-Basic classes. The mixin will define `__str__`, so we should probably rename `__str__` to `_sympystr`.

Reply:
This will also allow us to do the right thing by recursively calling `printer.doprint`.

Reply:
This is quite a can of worms, and something I'd prefer to handle in a file up patch. There are whole subpackages using wierd patterns like this.